lntest/node.go

```diff
@@ -1079,50 +1079,30 @@ func (hn *HarnessNode) WaitForNetworkChannelClose(ctx context.Context,
 	}
 }
 
-// WaitForBlockchainSync will block until the target nodes has fully
-// synchronized with the blockchain. If the passed context object has a set
-// timeout, then the goroutine will continually poll until the timeout has
-// elapsed. In the case that the chain isn't synced before the timeout is up,
-// then this function will return an error.
+// WaitForBlockchainSync waits for the target node to be fully synchronized with
+// the blockchain. If the passed context object has a set timeout, it will
+// continually poll until the timeout has elapsed. In the case that the chain
+// isn't synced before the timeout is up, this function will return an error.
 func (hn *HarnessNode) WaitForBlockchainSync(ctx context.Context) error {
-	errChan := make(chan error, 1)
-	retryDelay := time.Millisecond * 100
+	ticker := time.NewTicker(time.Millisecond * 100)
+	defer ticker.Stop()
 
-	go func() {
-		for {
-			select {
-			case <-ctx.Done():
-			case <-hn.quit:
-				return
-			default:
-			}
-
-			getInfoReq := &lnrpc.GetInfoRequest{}
-			getInfoResp, err := hn.GetInfo(ctx, getInfoReq)
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("timeout while waiting for " +
+				"blockchain sync")
+		case <-hn.quit:
+			return nil
+		case <-ticker.C:
```

Comment:
This is somewhat different as it won't wait 100 ms between `GetInfo` requests but will wait `100 ms - duration(GetInfo)`.

Reply:
Agreed (I think you meant `100ms + duration(GetInfo)`).
Unless we fire the goroutine under `case <-ticker.C`, or in the original code, under `case <-time.After(retryDelay)`, the time to wait between `GetInfo` will always be `100ms + duration(GetInfo)`.

Reply:
Nice, using the `NewTicker` will also avoid the timer leak that happens when the timer is not fired. One thing where the old more complicated solution was better is that it immediately returned if `hn.quit` was closed regardless if `GetInfo`  was blocking the goroutine but I think overall that is not a big issue and your design is cleaner.